velox/exec/tests/utils/HiveConnectorTestBase.cpp

```diff
@@ -63,35 +63,43 @@ void HiveConnectorTestBase::resetHiveConnector(
 
 void HiveConnectorTestBase::writeToFile(
     const std::string& filePath,
-    RowVectorPtr vector) {
-  writeToFile(filePath, std::vector{vector});
+    RowVectorPtr vector,
+    dwio::common::FileFormat fileFormat) {
+  writeToFile(filePath, std::vector{vector}, {}, fileFormat);
 }
 
 void HiveConnectorTestBase::writeToFile(
     const std::string& filePath,
     const std::vector<RowVectorPtr>& vectors,
-    std::shared_ptr<dwrf::Config> config) {
-  writeToFile(filePath, vectors, std::move(config), vectors[0]->type());
+    dwio::common::WriterOptions options,
+    dwio::common::FileFormat fileFormat) {
+  writeToFile(filePath, vectors, options, vectors[0]->type(), fileFormat);
 }
 
 void HiveConnectorTestBase::writeToFile(
     const std::string& filePath,
     const std::vector<RowVectorPtr>& vectors,
-    std::shared_ptr<dwrf::Config> config,
-    const TypePtr& schema) {
-  velox::dwrf::WriterOptions options;
-  options.config = config;
-  options.schema = schema;
+    dwio::common::WriterOptions options,
+    const TypePtr& schema,
+    dwio::common::FileFormat fileFormat) {
   auto localWriteFile = std::make_unique<LocalWriteFile>(filePath, true, false);
   auto sink = std::make_unique<dwio::common::WriteFileSink>(
       std::move(localWriteFile), filePath);
   auto childPool = rootPool_->addAggregateChild("HiveConnectorTestBase.Writer");
   options.memoryPool = childPool.get();
-  facebook::velox::dwrf::Writer writer{std::move(sink), options};
+  options.schema = schema;
+
+  // Defaults to DWRF writer if fileFormat is not currently supported yet.
+  if (!hasWriterFactory(fileFormat)) {
```

Comment:
Why do you need this? You could assume that a factory for the file format exist that is passed down. The test at the beginning wouldn't call this with parquet if the factory was not present in the test.
If the writer type is not present you get an appropriate test error.